libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/MainMenu.tsx

```diff
@@ -183,21 +187,48 @@ export class MainMenu extends React.PureComponent<
     );
   };
 
-  private toggleCohortSettingsPanel = (): void =>
+  private toggleCohortSettingsPanel = (): void => {
+    if (!this.state.cohortSettingsPanelVisible) {
+      this.props.telemetryHook?.({
+        level: TelemetryLevels.ButtonClick,
+        type: TelemetryEventName.MainMenuCohortSettingsClick
+      });
+    }
     this.setState((prev) => ({
       cohortSettingsPanelVisible: !prev.cohortSettingsPanelVisible
     }));
+  };
 
-  private toggleDashboardSettings = (): void =>
+  private toggleDashboardSettings = (): void => {
+    if (!this.state.dashboardSettingsVisible) {
+      this.props.telemetryHook?.({
```

Comment:
nit: maybe move these common calls into a small helper method where you only need to pass the type as a parameter to?

Reply:
Thanks for the comment, updated!